src/aura/CRLP_Rollup/CRLP_RollupController.js

```diff
@@ -110,11 +113,16 @@
      */
     onSave: function(cmp, event, helper){
         var activeRollup = cmp.get("v.activeRollup");
-        var canSave = helper.validateFields(cmp);
-        if(canSave){
+        if (cmp.get("v.mode") == 'delete') {
+            helper.toggleModal(cmp);
             helper.saveRollup(cmp, activeRollup);
-            cmp.set("v.mode", 'view');
-            helper.updateRollupName(cmp);
+        } else {
```

Comment:
Does it make sense to combine these 3 lines into one?
`} else if (helper.validateFields(cmp) === true) {`